code/ui/components/src/new/Select/Select.tsx

```diff
@@ -0,0 +1,180 @@
+import * as React from 'react';
+import * as SelectPrimitive from '@radix-ui/react-select';
+import { styled } from '@storybook/theming';
+import ExpandAlt from './icons/ExpandAlt';
+import Arrowup from './icons/Arrowup';
+import Arrowdown from './icons/Arrowdown';
+import Check from './icons/Check';
+
+const SelectTrigger = React.forwardRef<
+  React.ElementRef<typeof SelectPrimitive.Trigger>,
+  React.ComponentPropsWithoutRef<typeof SelectPrimitive.Trigger>
+>(({ className, children, ...props }, ref) => (
+  <StyledTrigger ref={ref} {...props}>
+    {children}
+    <SelectPrimitive.Icon asChild>
+      <ExpandAlt size={12} />
+    </SelectPrimitive.Icon>
+  </StyledTrigger>
+));
+SelectTrigger.displayName = SelectPrimitive.Trigger.displayName;
+
+const SelectContent = React.forwardRef<
+  React.ElementRef<typeof SelectPrimitive.Content>,
+  React.ComponentPropsWithoutRef<typeof SelectPrimitive.Content>
+>(({ className, children, ...props }, ref) => (
+  <SelectPrimitive.Portal>
+    <StyledContent ref={ref} {...props}>
+      <StyledScrollUpButton>
+        <Arrowup size={12} />
+      </StyledScrollUpButton>
+      <StyledViewport>{children}</StyledViewport>
+      <StyledScrollDownButton>
+        <Arrowdown size={12} />
+      </StyledScrollDownButton>
+    </StyledContent>
+  </SelectPrimitive.Portal>
+));
+SelectContent.displayName = SelectPrimitive.Content.displayName;
+
+const SelectLabel = React.forwardRef<
+  React.ElementRef<typeof SelectPrimitive.Label>,
+  React.ComponentPropsWithoutRef<typeof SelectPrimitive.Label>
+>(({ className, ...props }, ref) => <SelectPrimitive.Label ref={ref} {...props} />);
+SelectLabel.displayName = SelectPrimitive.Label.displayName;
+
+const SelectItem = React.forwardRef<
+  React.ElementRef<typeof SelectPrimitive.Item>,
+  React.ComponentPropsWithoutRef<typeof SelectPrimitive.Item>
+>(({ className, children, ...props }, ref) => (
+  <StyledItem ref={ref} {...props}>
+    <StyledItemIndicator>
+      <Check size={12} />
+    </StyledItemIndicator>
+    <SelectPrimitive.ItemText>{children}</SelectPrimitive.ItemText>
+  </StyledItem>
+));
+SelectItem.displayName = SelectPrimitive.Item.displayName;
+
+const SelectSeparator = React.forwardRef<
+  React.ElementRef<typeof SelectPrimitive.Separator>,
+  React.ComponentPropsWithoutRef<typeof SelectPrimitive.Separator>
+>(({ className, ...props }, ref) => <SelectPrimitive.Separator ref={ref} {...props} />);
+SelectSeparator.displayName = SelectPrimitive.Separator.displayName;
+
+export const Select = {
+  Root: SelectPrimitive.Root,
+  Group: SelectPrimitive.Group,
+  Value: SelectPrimitive.Value,
+  Trigger: SelectTrigger,
+  Content: SelectContent,
+  Label: SelectLabel,
+  Item: SelectItem,
+  Separator: SelectSeparator,
+};
```

Comment:
For complex combination-components like these `JSdoc` comments with mini examples would be great.

That way when users import these, all they have to do to get a example snippet, is hover over this import.

Reply:
This is just a wrapper on top of Radix with styles added to it. I can improve the docs to specify that I guess to make sure they go to Radix to understand all the use cases. We will not try to recreate their docs.